delayedbilling.php

```diff
@@ -218,6 +235,32 @@ function delayedbilling_civicrm_pre($op, $objectName, $id, &$params) {
   }
 }
 
+/**
+ * Implements hook_civicrm_post().
+ */
+function delayedbilling_civicrm_post($op, $objectName, $objectId, &$objectRef) {
+  if ($objectName == 'Contribution' && $op == 'edit') {
+    // Check to see if status == Failed.
+    $failedStatus = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Failed');
+    if ($objectRef->contribution_status_id == $failedStatus) {
+      // Check if this was a delayed payment.
+      if (_checkDelayedPayment($objectRef->contribution_page_id)) {
```

Comment:
Should we also check for the presence of a contribution_recur_id here to avoid the once off failures?